clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4569,7 +4569,7 @@ public ListTransactionsResult listTransactions(ListTransactionsOptions options)
     public FenceProducersResult fenceProducers(Collection<String> transactionalIds, FenceProducersOptions options) {
         AdminApiFuture.SimpleAdminApiFuture<CoordinatorKey, ProducerIdAndEpoch> future =
             FenceProducersHandler.newFuture(transactionalIds);
-        FenceProducersHandler handler = new FenceProducersHandler(logContext);
+        FenceProducersHandler handler = new FenceProducersHandler(logContext, requestTimeoutMs);
```

Comment:
There's a requestTimeoutMs in options, can you use that if specified?

Reply:
as an override to the AdminClient REQUEST_TIMEOUT_MS_CONFIG ?
The options doc says it's on override on API timeout 
but yes we could use that too

Reply:
I looked at it, and the option timeoutMs is used already as an API timeout - ie to calc a deadline when retrying.
The requestTimeoutMs is instead used for a single append, so I'd actually prefer not to mix the two and leave the PR as is.

Reply:
https://issues.apache.org/jira/browse/KAFKA-16047?focusedCommentId=17851019

Reply:
Hi @gharris1727 I'd appreciate if you can take a 2nd look or approve as IMHO without this small fix, exactly once MM2/connect is unusable in production

Reply:
If the AdminClient is no longer waiting for the fence request to complete, does it make sense for the underlying append to continue?

If the AdminClient is willing to wait longer for the fence request to complete, should the underlying append be aborted early?

If someone wants to raise the timeout for this one operation, I don't think that we should require them to increase the client-global request.timeout.ms.